crates/sui-framework/sources/object.move

```diff
@@ -127,6 +127,12 @@ module sui::object {
     /// restrictable in the object's module.
     native fun borrow_uid<T: key>(obj: &T): &UID;
 
+    // === one-time witness ===
+
+    /// Tests if the argument type is a one-time witness, that is a type with only one instantiation
+    /// across the entire code base.
+    public native fun is_one_time_witness<T: drop>(_: &T): bool;
```

Comment:
Discussed this with @awelc on Friday. I think it is fine to make this `(_: T): bool`. 
The reasoning is as follows:

The main use-case for this check is you want to ensure some function is called _at most once_ for a given type. To ensure this, the function that calls `is_one_time_witness` will need to take the witness type as an owned value anyway, to ensure it consumes the witness. 
In other words, a function that takes a one-time witness by reference can always be mis-used and called more than once.